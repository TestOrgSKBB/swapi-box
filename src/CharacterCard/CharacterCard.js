import React, { Component } from 'react';
import './CharacterCard.scss'
import favoriteImg from '../images/favorite.png';
import activeFavoriteImg from '../images/active-favorite.png';

class Character extends Component {

  toggleFavorited = () => {
    const { name } = this.props.character;
    this.props.updateFavorite(name);
  }

  returnFilms = () => {
    const films = this.props.character.films;
    return films.map((film, i) => <p className="character__card--films" key={i}>{film.title}</p>);
  }

  render() {
    const {name, homeworld, species, isFavorited} = this.props.character;

    return (
      <article className="character__cards">
          { !isFavorited && <img 
          onClick={this.toggleFavorited} 
          className='article__favorite-button' 
          src={favoriteImg} 
          alt='favorite button' /> }
        { isFavorited && <img 
          onClick={this.toggleFavorited} 
          className='article__favorite-button' 
          src={activeFavoriteImg} 
          alt='favorite button' /> }
        <h5 className="character__card--element">Name: <span className="character__card--info">{name}</span></h5>
        <h5 className="character__card--element">HomeWorld: <span className="character__card--info">{homeworld[0]}</span></h5>
        <h6 className="character__card--element">Population: <span className="character__card--info">{homeworld[1]}</span></h6>
        <h5 className="character__card--element">Species: <span className="character__card--info">{species}</span></h5>
        <div>
          <h6 className="character__card--element">Featured Films:</h6>
          <div className="character__card--div">
          {this.returnFilms()}
          </div>
        </div>
      </article>
    );
  };
};

export default Character