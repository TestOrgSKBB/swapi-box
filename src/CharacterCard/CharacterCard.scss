$dark-grey: rgba(32, 37, 40, .8);
$card-background: rgba(145, 145, 145, 0.8);
$element-background: rgba(51, 51, 51, .5);
$red-font: #963025;

.character__cards {
  background: $card-background;
  border: 2px solid white;
  color: white;
  width: 26%;
  margin: 1% 1%;
  height: 57vh;
  text-align: center;
  position: relative;
  -webkit-transition: -webkit-transform .8s ease-in-out;
  transition: transform .1s ease-in-out;
}

.character__cards:hover {
  -webkit-transform: rotate(2deg);
  transform: rotate(2deg);
}

.character__card--element {
  background: $dark-grey;
  font-size: 1.5em;
  margin-top: 2%;
  margin-bottom: 2%;
}

.character__card--div {
  margin-top: 6%;
}

.character__card--films {
  font-size: 1em;
  margin-top:2%;
  margin-block-end: 0em;
  margin-block-start: 1%;
  background-color: $element-background;
  color:white;
}

.character__card--info {
  color: $red-font;
  font-size: 1.25em;
  text-shadow: 1px 0px $card-background;
}

.article__favorite-button {
  position: absolute;
  left: 0;
  max-width: 9%;
  margin-top: 2.4%;
  margin-left: 2%;
  -webkit-transition: -webkit-transform .8s ease-in-out;
  transition:         transform .3s ease-in-out;
}

.article__favorite-button:hover {
  cursor:pointer;
  // max-width: 12%;
  -webkit-transform: rotate(90deg);
  transform: rotate(90deg);
}
