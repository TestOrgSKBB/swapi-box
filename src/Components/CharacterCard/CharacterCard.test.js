import React from 'react';
import { shallow, mount } from 'enzyme';
import CharacterCard from './CharacterCard';

describe('CharacterCard', () => {
  let wrapper;
  const mockUpdateFavorite = jest.fn()
  const mockToggleFavorited = jest.fn()
  const films = ['Tv: The Movie', 'A Movie About Musicals'];
  const mockCharacter = {
    films,
    homeworld: 'Earth',
    name: 'Pants',
    species: 'Dog',
  };

  beforeEach(() => {
    wrapper = shallow(<CharacterCard
      key={100}
      character={mockCharacter}
      updateFavorite={mockUpdateFavorite}
      />)
  });

  it('should match the snapshot with all correct data', () => {
    expect(wrapper).toMatchSnapshot();
  });

describe('toggleFavorites', () => {
  it('should call updateFavorite', () => {
    wrapper = mount(<CharacterCard
      key={100}
      updateFavorite = {mockUpdateFavorite}
      toggleFavorited = {mockToggleFavorited}
      character={mockCharacter}
      />)
      wrapper.find('.article__favorite-button').simulate('click');
      expect(mockUpdateFavorite).toHaveBeenCalled()
    });
  });

  describe('toggleFavorited', () => {
    it('should toggle status of isFavorited when star icon is clicked', () => {
      wrapper.find('img').simulate('click');
      expect(mockUpdateFavorite).toHaveBeenCalledWith('Pants');
    });
  });

  describe('returnFilms', () => {
    it('should return the films a character is in, in a <p> tag.', () => {
      const films = wrapper.instance().returnFilms();
      expect(films.length).toEqual(2);
    })
  });
});

