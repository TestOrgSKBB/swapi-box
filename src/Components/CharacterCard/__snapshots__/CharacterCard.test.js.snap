// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot with all correct data 1`] = `
<article
  className="character__cards"
>
  <div
    className="favorite-button-container"
  >
    <img
      alt="favorite button"
      className="article__favorite-button"
      onClick={[Function]}
      src="favorite.png"
    />
  </div>
  <h5
    className="character__card--element"
  >
    Name: 
    <span
      className="character__card--info"
    >
      Pants
    </span>
  </h5>
  <h5
    className="character__card--element"
  >
    HomeWorld: 
    <span
      className="character__card--info"
    >
      E
    </span>
  </h5>
  <h6
    className="character__card--element"
  >
    Population: 
    <span
      className="character__card--info"
    >
      a
    </span>
  </h6>
  <h5
    className="character__card--element"
  >
    Species: 
    <span
      className="character__card--info"
    >
      Dog
    </span>
  </h5>
  <div>
    <h6
      className="character__card--element"
    >
      Featured Films:
    </h6>
    <div
      className="character__card--div"
    >
      <p
        className="character__card--films"
        key="0"
      />
      <p
        className="character__card--films"
        key="1"
      />
    </div>
  </div>
</article>
`;
